src/accountsdb/db.zig

```diff
@@ -1667,37 +1674,45 @@ pub const AccountsDB = struct {
     pub fn deleteAccountFiles(
         self: *Self,
         delete_account_files: []const FileId,
-    ) void {
+    ) !void {
+        const number_of_files = delete_account_files.len;
         defer {
-            const number_of_files = delete_account_files.len;
             self.stats.number_files_deleted.add(number_of_files);
         }
 
-        // hold the lock for the entire duration of delete to
-        // reduce how much contention exists with other threads.
-        self.file_map_safe.fd_keep_open_rw.lock();
-        defer self.file_map_safe.fd_keep_open_rw.unlock();
+        var account_file_infos = try std.ArrayList(struct { AccountFile, FileId }).initCapacity(
```

Comment:
Any reason to store the `FileId` again when the account file already carries its file id? Seems like we should just use that @0xNineteen

Reply:
great catch 68d5af6